Storage/SQL/BrowserDB.swift

```diff
@@ -203,65 +203,60 @@ open class BrowserDB {
             SentryIntegration.shared.sendWithStacktrace(message: "Unable to get a transaction: \(error.localizedDescription)", tag: "BrowserDB", severity: .error)
 
             // Check if the error we got is recoverable (e.g. SQLITE_BUSY, SQLITE_LOCK, SQLITE_FULL).
-            // If so, we *shouldn't* move the database file to a backup location and re-create it.
-            // Instead, just crash so that we don't lose any data.
+            // If so, we will try moving the database file to a backup location and copy it back to
+            // its original location. This should effectively "unlock" the database file.
```

Comment:
Hmm.. ok... I tested this in simulator by opening a `BEGIN EXCLUSIVE` on the DB using `sqlite3` in Terminal prior to launching the app when a schema upgrade is needed. This resulted in a LOCK error which then moved the DB, copied it back and continued with upgrading the schema correctly. If it's a FULL error, it should result in a `fatalError()` as before (although I agree we can probably try clearing up caches or something and retry). I wasn't clear on the distinction between BUSY and LOCK, but if you say there's a chance at corrupting things, then I guess I'll go back to the drawing board on this.

The good news though, is that I don't believe we'll even hit this code path hardly at all now. If no schema upgrade is needed, this code never executes anymore (unlike before).